src/cpp/fastdds/topic/TopicImpl.cpp

```diff
@@ -79,6 +79,18 @@ ReturnCode_t TopicImpl::check_qos(
             return ReturnCode_t::RETCODE_INCONSISTENT_POLICY;
         }
     }
+    if (qos.resource_limits().max_samples <
+            (qos.resource_limits().max_instances * qos.resource_limits().max_samples_per_instance))
+    {
+        logError(DDS_QOS_CHECK, "max_samples should be greater than max_instances * max_samples_per_instance");
+        return ReturnCode_t::RETCODE_INCONSISTENT_POLICY;
+    }
+    if ((qos.resource_limits().max_instances == 0 || qos.resource_limits().max_samples_per_instance == 0) &&
+            (qos.resource_limits().max_samples != 0))
+    {
+        logError(DDS_QOS_CHECK, "max_samples should be greater than max_instances * max_samples_per_instance");
```

Comment:
```suggestion
        logError(DDS_QOS_CHECK, "max_samples should be infinite when max_instances or max_samples_per_instance are infinite");
```

Reply:
Change implemented on commit:
a1574d0e0 Refs #14958: Adrian's suggestions: Change in logError to differentiate instance allocation consistency error cases